### ISSUE TITLE
[Data Views] Table layout: add `aria-sort` to column headings as appropriate

### ISSUE THREAD

**Issue by andrewhayward** [WordPress/gutenberg#56853]
## What problem does this address?

Columns in the table layout can be sorted, but this is only communicated visually, and not semantically. 

![Header cells from a table layout, reading "Title", "Author", "Status" and "Actions". "Author" has an upwards-pointing chevron to visually indicate it is sorted.](https://github.com/WordPress/gutenberg/assets/159848/73854ca7-a2ca-4512-8e5e-8643cad39e7c)

## What is your proposed solution?

If a column is sorted, [`aria-sort`](https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Attributes/aria-sort) should be set on the relevant `th`, with a value of `ascending` or `descending` as appropriate.

```html
...
  <thead>
    <tr>
      <th>
        <button ...>Title</button>
      </th>
      <th aria-sort="ascending">
        <button ...>Author ⌃</button>
      </th>
      <th>
        <button ...>Status</button>
      </th>
      <th>
        Author
      </th>
    </tr>
  </thead>
...
```
